### PR TITLE
header footer sidebar

### PR DESCRIPTION
**Pull request by jaortiz**

### CODE REVIEW

**Comment by wojtekmaj** [comment]
Is there any ETA for this?

File: components/header/src/overrides/header.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+// ==============================
+
+const headerStyles = (_, { fixed }) => {
+	const mq = useMediaQuery();
+	return mq({
+		label: getLabel('header'),
+		flex: 'none',
+		display: 'flex',
+		...(fixed && { height: ['calc(3.375rem + 1px)', null, 'calc(4.0625rem + 1px)'] }),
+	})[0];
```
Comment (wojtekmaj) [inline components/header/src/overrides/header.js:22]: One detail. Footer wrapper has white background color, while header wrapper has none - it's INNER wrapper that has background color applied. Consequently, when adding e.g. 

```js
const overrides = {
  Inner: {
    styles: (styles) => ({
      ...styles,
      maxWidth: 1280,
      margin: '0 auto',
    }),
  },
};
``` 

to both header and footer, footer is all white with nicely centered content, while header is transparent with white rectangle awkwardly sitting there.

image

File: components/header/src/overrides/inner.js [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+			zIndex: '1',
+			left: 0,
+			right: 0,
+			top: '100%',
+			overflow: 'hidden',
+			opacity: scrolled ? 0 : 1,
+			borderTop: '1px solid #e8e8ed',
+			transition: 'opacity .2s',
+			willChange: 'opacity',
+		},
```
Comment (wojtekmaj) [inline components/header/src/overrides/inner.js:44]: Why not simply border-bottom?

File: components/footer/src/overrides/footer.js [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+		position: 'relative',
+		overflow: 'hidden',
+		backgroundColor: '#fff',
+		marginRight: offsetSidebar && [null, null, null, '300px'],
+		'::before': {
+			content: '""',
+			display: 'block',
+			backgroundColor: COLORS.primary,
+			height: '1px',
+		},
```
Comment (wojtekmaj) [inline components/footer/src/overrides/footer.js:30]: Why not simply border-top?

File: components/footer/src/overrides/logoLink.js [hunk 4]
```diff
@@ -0,0 +1,39 @@
+/** @jsx jsx */
+
+import { jsx, useMediaQuery, getLabel } from '@westpac/core';
```
Comment (wojtekmaj) [inline components/footer/src/overrides/logoLink.js:3]: 'jsx' is defined but never used

File: components/header/src/Header.js [hunk 5]
```diff
@@ -0,0 +1,227 @@
...
+
+export const Header = ({
+	logoLink,
+	logoOnClick,
+	logoCenter,
+	logoAssistiveText,
+	leftIcon,
+	leftOnClick,
+	leftAssistiveText,
+	fixed,
```
Comment (wojtekmaj) [inline components/header/src/Header.js:30]: 'logoOnClick' and 'fixed' are missing in propTypes

**Comment by jaortiz** [comment]
> Is there any ETA for this?

@wojtekmaj I don't currently have an ETA as I am waiting on some ui/ux design reviews that need to be done as part of the upgrade from our old system. If this is a blocker for you I am fine to do some fixes and release this as an alpha version.

**Comment by wojtekmaj** [comment]
@jaortiz Not totally a blocker, to be completely honest I just copied the code over and it's working just fine :joy: But would be nice to know.

File: components/header/src/overrides/inner.js [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+
+	return mq({
+		label: getLabel('header-inner'),
+		position: 'relative',
+		flex: 1,
+		backgroundColor: '#fff',
+		textAlign: 'left',
+		marginLeft: 'auto',
+		marginRight: 'auto',
+		marginBottom: '1px',
```
Comment (wojtekmaj) [inline components/header/src/overrides/inner.js:26]: You left this to compensate for space taken by pseudoelement which has been recently removed and replaced by a simple border-bottom.
```suggestion
```

File: components/header/src/overrides/leftBtn.js [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+import { Button } from '@westpac/button';
+
+// ==============================
+// Component
+// ==============================
+
+const LeftBtn = ({ state: { leftIcon, leftOnClick, leftAssistiveText }, ...rest }) => {
+	// Can I make this cleaner
+	const Icon =
+		leftIcon === 'arrow' ? ArrowLeftIcon : leftIcon === 'hamburger' ? HamburgerMenuIcon : null;
```
Comment (wojtekmaj) [inline components/header/src/overrides/leftBtn.js:14]: I think you could…
```suggestion
	const Icon = (() => {
	  switch (leftIcon) {
		case 'arrow': return ArrowLeftIcon;
		case 'hamburger': return HamburgerMenuIcon;
		default: return null
	  }
	})();
```

**Comment by wojtekmaj** [comment]
Feature request: make logo on the right side optional

**Comment by wojtekmaj** [comment]
@jaortiz @jonnystening Are there plans to merge this anytime soon, preferably with the option to disable the footer logo?

**Comment by jaortiz** [comment]
@wojtekmaj so sorry about the delay on this. Ive finally gotten the approval to release this, I will release it tomorrow.

**Pull request merged**